src/sema/ast.rs

```diff
@@ -638,6 +638,13 @@ impl Contract {
             })
             .cloned()
     }
+
+    /// Does this contract have at least one public function
+    pub fn has_public_functions(&self, ns: &Namespace) -> bool {
+        self.functions
+            .iter()
+            .any(|func_no| ns.functions[*func_no].is_public())
+    }
```

Comment:
This should check for function type and use all functions, included inherited ones.

```rust
    pub fn has_public_function(&self, ns: &Namespace) -> bool {
        self.all_functions.keys().any(|func_no| {
            let func = &ns.functions[*func_no];

            func.ty == pt::FunctionTy::Function && func.is_public()
        })
    }
```